[distributed] Enable H100 test for all distributed related changes

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #156721

We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc @H-Huang @awgu @wanchaol @fegin @wz337 @wconstab @d4l3k

fduwjj committed:
[distributed] Enable H100 test for all distributed related changes

[ghstack-poisoned]

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]

huydhn commented on .github/label_to_label.yml:
```diff
@@ -48,3 +48,12 @@
+  - "release notes: distributed (dtensor)"
+  - "oncall: distributed"
+  then:
+  - "ciflow/h100-distributed"
```
cc @clee2000 if you have any thoughts here, maybe this is only meant to be used on issue

clee2000 replied:
This file is only used for issues

fduwjj replied:
hmmm, if this is the case what is the recommended way for PR? @clee2000 @huydhn

huydhn replied:
I think it's referred to the logic here https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/autoLabelBot.ts#L149-L155

fduwjj commented on .github/workflows/h100-distributed.yml:
```diff
@@ -1,13 +1,16 @@
       - ciflow/h100-distributed/*
+  workflow_dispatch:
+  schedule:
+    - cron: 46 8 * * *  # about 1:46am PDT
```
to my understanding not every PR triggers this CI so we have a daily run everyday? This is mimic what trunk is doing.

fduwjj replied:
oh that will be too much since we only want to run this CI for distributed related changes. that's why we have the change regarding the labeling. Do you have any better suggestions how we should approach this? Thanks.

huydhn replied:
Oh, if you want to run this on every trunk commit, then just removing `branches` and keep the schedule to turn this into a periodic job

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]

fduwjj committed:
Update on "[distributed] Enable H100 test for all distributed related changes"


We want to run H100 CI for distributed related changes. We already have a labeling of oncall:distributed when touching distributed related code: https://github.com/pytorch/pytorch/blob/4491326fb0c0e67eca1598ae33c41cdfced2cd33/.github/labeler.yml#L94. So we want to leverage that.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k

[ghstack-poisoned]
